src/utils/flash.js

```diff
@@ -282,36 +249,37 @@ export function useFastboot() {
         setProgress(0)
 
         async function flashDevice() {
-          const currentSlot = await fastboot.current.getVariable('current-slot')
+          const currentSlot = await qdl.current.getActiveSlot();
           if (!['a', 'b'].includes(currentSlot)) {
             throw `Unknown current slot ${currentSlot}`
           }
+          const otherSlot = currentSlot === 'a' ? 'b' : 'a'
+
+          // Erase current xbl partition so if users try to power up device
+          // with corrupted primary gpt header, it would not update the backup
```

Comment:
can you explain this more?

Reply:
Whenever booting up the device, the xbl partition will try to compare the primary gpt headers with their corresponding backup gpt headers, and if the backups are not the same as the primary gpt headers, it will update the backup to be the primary gpt headers.

If a user somehow disconnect the device during changing active slot process then the device would have corrupted gpt header, and if the user tries to boot up the device immediately afterwards without flashing again, the xbl would just update the backup gpt headers with the corrupted primary one. Erasing that makes sure that that doesn't happen.

 it is important to always keep the backup gpt headers clean and uncorrupted since we use those to recover the corrupted gpt headers if there were any failure during flashing process. 
 
 Fastboot doesn't have this feature, and it would just go with the corrupted gpt headers (tested).